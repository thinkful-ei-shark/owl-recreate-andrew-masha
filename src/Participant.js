import React from 'react';

function Participant(props) {
  return (
    <div className="Participant" key={props.id}>
      <div className="portait">
        <img src={props.avatar} alt="avatar" />
      </div>
      <div className="participant-meta">
        <div className="participant-name">
          {props.name}
        </div>
        <div className="Participant-byline">
          <span 
            className={`dot ${props.inSession ? 'in-session' : 'not-in-session'}`}>
          </span>
          <span className="stage-status">
            {`${props.onStage ? 'On Stage' : 'Off Stage'}`}
          </span>
        </div>
      </div>
    </div>
  )
}

export default Participant;