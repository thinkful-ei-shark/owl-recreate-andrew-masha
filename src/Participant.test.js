import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Participant from './Participant.js'

const testObject = {
  id: 1,
  name: 'Koren Templeton',
  avatar:
      'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
  inSession: true,
  onStage: true
}

it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(
    <Participant props={testObject} />,
    div
  )

  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<Participant props={testObject} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});