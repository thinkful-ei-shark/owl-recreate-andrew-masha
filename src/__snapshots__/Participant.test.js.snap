// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Participant"
>
  <div
    className="portait"
  >
    <img
      alt="avatar"
    />
  </div>
  <div
    className="participant-meta"
  >
    <div
      className="participant-name"
    />
    <div
      className="Participant-byline"
    >
      <span
        className="dot not-in-session"
      />
      <span
        className="stage-status"
      >
        Off Stage
      </span>
    </div>
  </div>
</div>
`;
